src/components/ListItem.jsx

```diff
@@ -1,5 +1,69 @@
+import Checkbox from '@mui/material/Checkbox';
+import { useState, useEffect } from 'react';
 import './ListItem.css';
+import { updateItem } from '../api';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({ item }) {
+	// Destructure item props
+	const { name, dateLastPurchased } = item;
+
+	// Get the initial checked state from localStorage or default to false
+	const [checked, setChecked] = useState(() => {
+		const storedChecked = localStorage.getItem(`checked-${name}`);
+		return storedChecked ? JSON.parse(storedChecked) : false;
+	});
+
+	// Function to check if 24 hours have passed
+	const has24HoursPassed = (dateLastPurchased) => {
+		const purchaseTime = dateLastPurchased.getTime(); // Time in milliseconds
+		const currentTime = new Date().getTime(); // Current time in milliseconds
+		const oneDayInMs = 24 * 60 * 60 * 1000; // 24 hours in milliseconds
```

Comment:
**nit (non-blocking):** as this is a constant, consider renaming in uppercase snake case. Note: this constant exists in `dates.js`